ivy/functional/frontends/paddle/tensor/tensor.py

```diff
@@ -908,3 +908,51 @@ def gather(self, y, name=None):
     def gather_(self, y, name=None):
         res = self.gather(self, y)
         return ivy.inplace_update(self, res)
+
+    def _fill_diagonal_tensor_impl(self, x, y, offset=0, dim1=0, dim2=1, inplace=False):
+        inshape = x.shape
+        print(inshape)
+        assert dim1 < len(inshape) and dim1 >= -len(
+            inshape
+        ), "dim1 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert dim2 < len(inshape) and dim2 >= -len(
+            inshape
+        ), "dim2 should be between [-rank, rank) in fill_diagonal_tensor_"
+        assert len(inshape) >= 2, "Tensor dims should be >= 2 in fill_diagonal_tensor_"
+        dim1 %= len(inshape)
+        dim2 %= len(inshape)
```

Comment:
Are you sure that is computationally correct? Why don't you use more simpler terms?